addons/misra.py

```diff
@@ -3964,12 +3964,16 @@ def misra_21_5(self, data):
             self.reportError(directive, 21, 5)
 
     def misra_21_6(self, data):
-        dir_stdio = findInclude(data.directives, '<stdio.h>')
-        dir_wchar = findInclude(data.directives, '<wchar.h>')
-        if dir_stdio:
-            self.reportError(dir_stdio, 21, 6)
-        if dir_wchar:
-            self.reportError(dir_wchar, 21, 6)
+        for token in data.tokenlist:
+            if not isFunctionCall(token) or token.previous.function:
+                continue
+            standard = data.standards.c
+            if ((standard == 'c89' and token.previous.str in C90_STDLIB_IDENTIFIERS["stdio.h"]) or
```

Comment:
this condition does not make sense. If `standard` is c89 we will still check if the funcname is in `C11_STDLIB_IDENTIFIERS["stdio.h"]`